source/extensions/filters/http/jwt_authn/data_store.h

```diff
@@ -0,0 +1,82 @@
+#pragma once
+
+#include "envoy/server/filter_config.h"
+#include "envoy/thread_local/thread_local.h"
+
+#include "common/common/logger.h"
+
+#include "extensions/filters/http/jwt_authn/extractor.h"
+#include "extensions/filters/http/jwt_authn/jwks_cache.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace JwtAuthn {
+
+/**
+ * The object to store config and caches. It has following objects:
+ * * config_: the config.
+ * * jwks_cache_ for Jwks cache.
+ * * extractor_ for token extraction
+ * In the future, it will have token cache.
+ * It is per-thread and stored in thread local.
+ */
+class DataStore : public ThreadLocal::ThreadLocalObject {
+public:
+  // Load the config from envoy config.
+  DataStore(const ::envoy::config::filter::http::jwt_authn::v2alpha::JwtAuthentication& config)
+      : config_(config), jwks_cache_(JwksCache::create(config_)),
+        extractor_(Extractor::create(config_)) {}
+
+  // Get the Config.
+  const ::envoy::config::filter::http::jwt_authn::v2alpha::JwtAuthentication& config() const {
+    return config_;
+  }
+
+  // Get the JwksStore object.
+  JwksCache& getJwksCache() { return *jwks_cache_; }
+
+  // Get the token  extractor.
+  const Extractor& getExtractor() const { return *extractor_; }
+
+private:
+  // Store the config.
+  const ::envoy::config::filter::http::jwt_authn::v2alpha::JwtAuthentication& config_;
+  // The JwksCache object.
+  JwksCachePtr jwks_cache_;
+  // The object to extract tokens.
+  ExtractorConstPtr extractor_;
+};
+
+// The factory to create per-thread auth store object.
```

Comment:
Is there a better name than factory here? E.g. `ThreadLocalDataStore`?

Reply:
Rename it to Config